tests/cuda_ndarrays/conftest.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=arguments-differ, inconsistent-return-statements, protected-access, abstract-method/
+import subprocess
+import os
+import pathlib
+import sys
+import shutil
+import pytest
+
+NEEDS_FROM_PARENT = hasattr(pytest.Item, "from_parent")
+
+def pytest_collect_file(parent, path):
+    """
+    A hook to collect test_*.cu test files.
+
+    """
+    if path.ext == ".cu" and path.basename.startswith("test_"):
+        if NEEDS_FROM_PARENT:
+            return CTestFile.from_parent(path=pathlib.Path(path), parent=parent)
+        return CTestFile(parent=parent, path=pathlib.Path(path))
+
+def pytest_collection_modifyitems(items):
+    """
+    a hook to modify the items before the tests for C Cuda unit test files.
+
+    """
+    for item in items:
+        if item.fspath.ext == ".cu":
+            item._nodeid = item.nodeid + " < " + item.test_result["DSCR"] + " >"
+
+class CTestFile(pytest.File):
+    """
+    A custom file handler class for C unit test files.
+
+    """
+
+    @classmethod
+    def from_parent(cls, **kwargs):
+        return super().from_parent(**kwargs)
+
+    def collect(self):
+        """
+        Overridden collect method to collect the results from each
+        C unit test executable.
+
+        """
+        # Run the exe that corresponds to the .c file and capture the output.
+        test_exe = os.path.splitext(str(self.fspath))[0]
+        rootdir = str(self.config.rootdir)
+        test_exe = os.path.relpath(test_exe)
+        ndarray_path =  os.path.join(rootdir , "pyccel", "stdlib", "ndarrays")
+        cuda_ndarray_path = os.path.join(rootdir, "pyccel", "stdlib", "cuda_ndarrays")
+        comp_cmd = [shutil.which("nvcc"), test_exe + ".cu",
+                    os.path.join(ndarray_path, "ndarrays.c"),
+                    os.path.join(cuda_ndarray_path, "cuda_ndarrays.cu"),
+                    "-I", ndarray_path,
+                    "-I", cuda_ndarray_path,
+                    "-o", test_exe,]
+
+        subprocess.run(comp_cmd, check= 'TRUE')

```

Comment:
Are you sure the `check` argument takes a string?

Reply:
Sorry, should be a boolean not a string